### PR TITLE
Workload reset test case

### PR DESCRIPTION
**Pull request by nikhil-achole**

### CODE REVIEW

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    @classmethod
+    def setup_clients(cls):
+        super(WorkloadTest, cls).setup_clients()
+        cls.client = cls.os.wlm_client
+        reporting.add_test_script(str(__name__))
+    
+    def test_workload_reset(self):
+        try:
+            global volumes
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: Is this required?

Comment (nikhil-achole) [reply tempest/api/workloadmgr/workload/test_workload_reset.py]: no removed

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                    reporting.add_test_step("Create workload-{}".format(i), tvaultconf.FAIL)
+                    reporting.set_test_script_status(tvaultconf.FAIL)
+            else:
+                reporting.add_test_step("Create workload-{}".format(i), tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+                raise Exception("Workload creation failed")
+
+            ### Full snapshot ###
+
+            self.created = False
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: Is this variable used ? If not, remove it.

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+            i+=1
+            volsnaps_before = self.get_volume_snapshots(volume_id)
+
+            LOG.debug("\nVolume snapshots after full snapshot : {}\n".format(volsnaps_before))
+           
+            self.workload_reset(workload_id)
+ 
+            
+            self.created = False
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: Is this variable used ? If not, remove it.

Comment (nikhil-achole) [reply tempest/api/workloadmgr/workload/test_workload_reset.py]: done

File: tempest/api/workloadmgr/workload/test_workload_reset.py (hunk 2 above)
Comment (nikhil-achole) [reply tempest/api/workloadmgr/workload/test_workload_reset.py]: done

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                reporting.add_test_step("Create full snapshot-{}".format(i), tvaultconf.PASS)
+                LOG.debug("Full snapshot available!!")
+            else:
+                reporting.add_test_step("Create full snapshot-{}".format(i), tvaultconf.FAIL)
+                raise Exception("Snapshot creation failed")
+
+
+            volsnaps_after = self.get_volume_snapshots(volume_id)
+            LOG.debug("\nVolume snapshots after full snapshot : {}\n".format(volsnaps_after))
+
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: What is the verification step for workload reset ?

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            time.sleep(5)
+            self.wait_for_workload_tobe_available(workload_id)
+            if(self.getSnapshotStatus(workload_id, snapshot_id_1) == "available"):
+                reporting.add_test_step("Create full snapshot-{}".format(i), tvaultconf.PASS)
+                LOG.debug("Full snapshot available!!")
+            else:
+                reporting.add_test_step("Create full snapshot-{}".format(i), tvaultconf.FAIL)
+                raise Exception("Snapshot creation failed")
+
+            import pdb;pdb.set_trace()
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: Remove pdb statement

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            resp, body = self.wlm_client.client.get("/workloads/"+workload_id+"/snapshots/"+snapshot_id_1)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: You can use existing methods from base.py to fetch snapshot details

File: tempest/api/workloadmgr/workload/test_workload_reset.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            if body['snapshot']['snapshot_type'] == "full":
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload_reset.py]: Please provide more details on this workload reset verification step.

**Pull request merged**